src/store/model/helpers/nvidia.ts

```diff
@@ -1,89 +1,73 @@
-import {Browser, Response} from 'puppeteer';
+import {Browser, Page, Response} from 'puppeteer';
 import {NvidiaRegionInfo, regionInfos} from '../nvidia-api';
 import {Config} from '../../../config';
 import {Link} from '../store';
 import {Logger} from '../../../logger';
 import open from 'open';
 import {timestampUrlParameter} from '../../timestamp-url-parameter';
 
-const nvidiaApiKey = '9485fa7b159e42edb08a83bde0d83dia';
-
 function getRegionInfo(): NvidiaRegionInfo {
 	const country = Array.from(regionInfos.keys()).includes(Config.store.country) ? Config.store.country : 'usa';
 
-	const defaultRegionInfo: NvidiaRegionInfo = {drLocale: 'en_us', fe2060SuperId: 5379432500, fe3080Id: 5438481700, fe3090Id: null, nvidiaLocale: 'en_us'};
+	const defaultRegionInfo: NvidiaRegionInfo = {currency: 'USD', drLocale: 'en_us', fe2060SuperId: 5379432500, fe3080Id: 5438481700, fe3090Id: null, nvidiaLocale: 'en_us'};
 	return regionInfos.get(country) ?? defaultRegionInfo;
 }
 
-function digitalRiverStockUrl(id: number, drLocale: string): string {
-	return `https://api.digitalriver.com/v1/shoppers/me/products/${id}/inventory-status?` +
-		`&apiKey=${nvidiaApiKey}` +
-		`&locale=${drLocale}` +
-		timestampUrlParameter();
+function nvidiaStockUrl(id: number, drLocale: string, currency: string): string {
+	return `https://api-prod.nvidia.com/direct-sales-shop/DR/products/${drLocale}/${currency}/${id}?` +
+		timestampUrlParameter().slice(1);
 }
 
 interface NvidiaSessionTokenJSON {
-	access_token: string;
+	session_token: string;
+}
+
+interface NvidiaAddToCardJSON {
+	location: string;
 }
 
 function nvidiaSessionUrl(nvidiaLocale: string): string {
 	return `https://store.nvidia.com/store/nvidia/SessionToken?format=json&locale=${nvidiaLocale}` +
-		`&apiKey=${nvidiaApiKey}` +
 		timestampUrlParameter();
 }
 
-function addToCartUrl(id: number, drLocale: string, token: string): string {
-	return 'https://api.digitalriver.com/v1/shoppers/me/carts/active/line-items?format=json&method=post' +
-		`&productId=${id}` +
-		`&token=${token}` +
-		'&quantity=1' +
-		`&locale=${drLocale}` +
-		timestampUrlParameter();
-}
+async function addToCartAndGetLocationRedirect(page: Page, sessionToken: string, productId: number): Promise<string> {
+	const url = 'https://api-prod.nvidia.com/direct-sales-shop/DR/add-to-cart';
+
+	page.removeAllListeners('request');
+
+	await page.setRequestInterception(true);
+
+	page.on('request', interceptedRequest => {
```

Comment:
Nice, this will speed this up quite a bit. Was looking for this type of API in puppeteer. Nice find!

Reply:
Sorry if I'm being dense, but what is the requested change?

Reply:
Oh.. I didn't send :o

Reply:
Lol I had two reviews open... Whoops. Thanks!